README.md

```diff
@@ -79,10 +96,75 @@ think that this is the natural way to define the schema, so we built our type sy
 will also allow developers to define the schema in formats native to different ecosystems, such as PHP array, YAML and 
 even XML.
 
-## Usage
+### Validation
+
+**Package lead: [@crisu83](https://github.com/crisu83/)**
+
+### Schema Validation
 
 TODO
 
+### Query validation
+
+Even though some GraphQL implementations come without any query validation we decided to include it in the first 
+version as it is a part of the specification. Query validation is done by evaluating the query AST using validation 
+rules. We implemented this using the [Visitor pattern](https://en.wikipedia.org/wiki/Visitor_pattern) very much like 
+it is implemented in the reference implementation. 
+
+In practice this means that the query AST is visited by the each rule. Every rule takes care of 
+validating the query against a particular part of the 
+[Validation specification](http://facebook.github.io/graphql/October2016/#sec-Validation). This also allows 
+developers to easily implement their own validation rules.
+
+## Usage
+
+### Installation
+
+This library cannot yet be installed through [Composer](https://getcomposer.org/) because it is still being developed. 
+As soon we have the first working version we will list it on [Packagist](https://packagist.org/). In the meanwhile you 
+can clone the project and try it out yourself. 
+
+### Example
+
+This script demonstrates the public API:
+
+```php
+use function Digia\GraphQL\graphql;
+
+// Assuming you have defined your types.
```

Comment:
This is not very useful. How do you define the types?

Reply:
I thought It would be too much code and you'd probably do that in a separate file.

Reply:
Fixed.